### PR TITLE
feat: Add liquidations functionality coverage ( SC-4448)

### PR DESCRIPTION
**Pull request by JGcarv**

### CODE REVIEW

File: contracts/test/ClaimTest.t.sol [hunk 1]
```diff
@@ -27,6 +33,12 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
     uint256 constant BTC = 10 ** 8;   // WBTC precision
     uint256 constant USD = 10 ** 6;   // USDC precision
 
+    uint8 constant INTEREST_ONLY = 1;
+    uint8 constant PARTIALLY_AMORTIZED = 2;
+    uint8 constant FULLY_AMORTIZED = 3;
+    uint8 constant NOT_COLLATERALIZED = 4;
+    uint8 constant UNDER_COLLATERALIZED = 5;
```
Comment (lucas-manuel) [inline contracts/test/ClaimTest.t.sol:40]: Alignment

File: contracts/test/ClaimTest.t.sol [hunk 2]
```diff
@@ -578,4 +621,281 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
...
+        assertEq(details[0], 280_135_620000);
+        assertEq(details[1], 0);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 280_135_620000);
+        assertEq(details[6], 719_864_380000);
+    }
+
+    function test_claim_liquidationDOS() external {
```
Comment (lucas-manuel) [inline contracts/test/ClaimTest.t.sol:783]: Lets make another copy of this test, but use stopLiquidation to stop the dos

File: contracts/test/ClaimTest.t.sol [hunk 3]
```diff
@@ -578,4 +621,281 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
...
+        assertEq(details[0], 280_135_620000);
+        assertEq(details[1], 0);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 280_135_620000);
+        assertEq(details[6], 719_864_380000);
+    }
+
+    function test_claim_liquidatonStop() external {
```
Comment (lucas-manuel) [inline contracts/test/ClaimTest.t.sol:830]: test_claim_liquidatonStop_before_liquidation

Comment (lucas-manuel) [reply contracts/test/ClaimTest.t.sol:830]: Should add a comment outlining that this is a very bad thing, and add some code to pull funds and show how the collateralAsset can be manually recovered after this

File: contracts/test/ClaimTest.t.sol [hunk 4]
```diff
@@ -578,4 +621,281 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
         // Fails to claim after loan is closed
         try pool.claim(address(loanV2), address(debtLockerFactory)) { assertTrue(false, "Able to claim"); } catch { }
     }
+
+    function test_claim_defaultUncollateralized() external {
```
Comment (lucas-manuel) [inline contracts/test/ClaimTest.t.sol:625]: For these more basic tests, we should actually assert that the entire BPT burning flow works

Comment (lucas-manuel) [reply contracts/test/ClaimTest.t.sol:625]: Just for the uncollateralized and undercollateralized basic tests. Once we assert that its working with the claim amounts we're fine for the rest of the tests

Comment (lucas-manuel) [reply contracts/test/ClaimTest.t.sol:625]: You can refer to the maple-core V1 integration tests for how to assert the BPT burning flow and LP losses

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Excellent PR, some added changes

File: contracts/test/ClaimTest.t.sol [hunk 5]
```diff
@@ -578,4 +621,411 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
...
+
+        assertEq(details[0], 280_135_620000);
+        assertEq(details[1], 0);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 280_135_620000);
+        assertEq(details[6], 719_864_380000);
+
+        assertEq(pool.principalOut(), pool_principalOut - 719_864_380000 - 280_135_620000); // Principal out reduced by full amount
```
Comment (lucas-manuel) [inline contracts/test/ClaimTest.t.sol:808]: Can be `1_000_000_000000`

File: contracts/test/ClaimTest.t.sol [hunk 6]
```diff
@@ -578,4 +621,411 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
...
```
Comment (lucas-manuel) [inline contracts/test/ClaimTest.t.sol:857]: Can be `1_000_000_000000`

File: contracts/test/ClaimTest.t.sol [hunk 7]
```diff
@@ -578,4 +621,411 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
...
+
+        assertEq(details[0], 0);
+        assertEq(details[1], 0);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 0);
+        assertEq(details[6], 1_000_000_000000);   
+
+        assertEq(pool.principalOut(), pool_principalOut - 1_000_000_000000);
```
Comment (lucas-manuel) [inline contracts/test/ClaimTest.t.sol:943]: Still need to demo pullFunds after a bad stopLiquidation

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: contracts/test/ClaimTest.t.sol (hunk 7 above)
Comment (JGcarv) [reply contracts/test/ClaimTest.t.sol:943]: It's on the test case below it!

**Pull request merged**